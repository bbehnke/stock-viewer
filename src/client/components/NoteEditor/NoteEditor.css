.note-editor {
  display: flex;
  flex: 1;
  width: 100%;
}

.note-editor-form {
  display: flex;
  flex-direction: column;
  flex: 1;
}

.note-editor-text-area {
  height: 400px;
  flex: 1;
  resize: none;
  font-size: 16px;
  border: 1px solid black;
}

.note-editor-submit {
  font-size: 16px;
}
