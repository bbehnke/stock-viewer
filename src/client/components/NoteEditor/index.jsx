import React from 'react';
import PropTypes from 'prop-types';
import './NoteEditor.css';

class NoteEditor extends React.Component {
  constructor() {
    super();
    this.state = {};
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  static getDerivedStateFromProps(props, state) {
    const { note } = props;
    const { id } = state;
    if (id === note.id) {
      return null;
    }
    return {
      id: note.id,
      value: note.value
    };
  }

  handleChange(event) {
    this.setState({
      value: event.target.value
    });
  }

  handleSubmit(event) {
    const { value } = this.state;
    const { onSaveClick, note } = this.props;
    onSaveClick(note.id, value);
    event.preventDefault();
  }

  render() {
    const { value } = this.state;
    return (
      <div className="note-editor">
        <form className="note-editor-form" onSubmit={this.handleSubmit}>
          <textarea
            className="note-editor-text-area"
            value={value}
            onChange={this.handleChange}
          />
          <button className="note-editor-submit" type="submit">Save note</button>
        </form>
      </div>
    );
  }
}

NoteEditor.propTypes = {
  note: PropTypes.shape({
    id: PropTypes.number.isRequired,
    value: PropTypes.string.isRequired
  }).isRequired,
  onSaveClick: PropTypes.func.isRequired
};

export default NoteEditor;
