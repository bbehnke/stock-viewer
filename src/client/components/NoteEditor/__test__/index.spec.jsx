import React from 'react';
import { mount } from 'enzyme';
import NoteEditor from '..';

describe('NoteEditor tests', () => {
  it('renders note editor', () => {
    const note = { id: 1, value: 'test_value' };
    const component = mount(<NoteEditor note={note} onSaveClick={() => {}} />);
    expect(component).toMatchSnapshot();
  });
  it('onSaveClick called', () => {
    const note = { id: 1, value: 'test_value' };
    const updatedValue = 'test_updated_value';
    const onSaveClickMock = jest.fn();
    const component = mount(<NoteEditor note={note} onSaveClick={onSaveClickMock} />);
    component.find('textarea.note-editor-text-area').simulate('change', { target: { value: updatedValue } });
    component.find('button.note-editor-submit').simulate('submit');
    expect(onSaveClickMock).toBeCalledWith(note.id, updatedValue);
  });
});
