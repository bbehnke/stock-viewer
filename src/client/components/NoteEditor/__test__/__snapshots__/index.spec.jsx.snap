// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteEditor tests renders note editor 1`] = `
<NoteEditor
  note={
    Object {
      "id": 1,
      "value": "test_value",
    }
  }
  onSaveClick={[Function]}
>
  <div
    className="note-editor"
  >
    <form
      className="note-editor-form"
      onSubmit={[Function]}
    >
      <textarea
        className="note-editor-text-area"
        onChange={[Function]}
        value="test_value"
      />
      <button
        className="note-editor-submit"
        type="submit"
      >
        Save note
      </button>
    </form>
  </div>
</NoteEditor>
`;
